compiler/src/iree/compiler/Codegen/Dialect/GPU/Transforms/Transforms.cpp

```diff
@@ -620,16 +620,16 @@ distributeMultiMmaOp(RewriterBase &rewriter, IREE::GPU::MultiMmaOp mmaOp,
       accStrides);
 
   // Step 3. Create the new multi_mma op.
-  auto newKind = mmaOp.getKind();
-  if (auto dataTiledMma = dyn_cast<DataTiledMMAAttr>(newKind)) {
-    newKind = DataTiledMMAAttr::get(
-        context, dataTiledMma.getIntrinsic(), dataTiledMma.getUnrollM(),
-        /*subgroups_m=*/1, dataTiledMma.getUnrollN(),
-        /*subgroups_n=*/1, dataTiledMma.getUnrollK());
-  }
```

Comment:
I did not pay much attention on the changes of DataTiledMMAAttr. Why do we drop the newKind here? Does it impact codegen path? Or it is handled by attribute interface implementation? Thanks for your explanation in advance, and perhaps we can put the information to the PR description.

Reply:
We are now transporting the old kind, unchanged.  The code being deleted here was creating a new kind that only preserved the `unroll_*` parameters, but had the `subgroups_*` parameters set to 1. I thought that those parameters were inherently not needed after thread-distribution.  That changed with ukernels. While it remains true that the `subgroups_*` parameters should not be needed after thread-distribution, in order to avoid needing them, codegen makes use of all the stride information for all the expanded dimensions. We could pass all these strides to the ukernels, but that woud be cumbersome, particularly as the number of dimension varies as unit dimensions are omitted. So in this case, passing the original DataTiledMMAAttr parameters and letting the ukernel infer the strides, results in much simpler code. The drawback is an interaction-at-a-distance in the layouts implied by these parameters, but I think that's OK.